internal/controller/pkg/resolver/reconciler.go

```diff
@@ -310,19 +310,45 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		format = packageDigestFmt
 	}
 
-	pack.SetSource(fmt.Sprintf(format, ref.String(), addVer))
+	tarSource := fmt.Sprintf(format, ref.String(), addVer)
```

Comment:
```suggestion
	desiredSource := fmt.Sprintf(format, ref.String(), addVer)
```

nit: `tarSource` reads like it a source file in `tar` format 😅